packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -965,6 +998,67 @@ if (y) {
       ],
     }),
 
+    // nullable enum in boolean context
+    {
+      options: [{ allowNullableEnum: false }],
+      code: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
+        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
+        if (theEnum) {
+        }
+      `,
+      errors: [
+        {
+          line: 7,
+          column: 13,
+          messageId: 'conditionErrorNullableEnum',
+          endLine: 7,
+          endColumn: 20,
+        },
+      ],
+      output: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
+        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
+        if (theEnum != null) {
+        }
+      `,
+    },
+    {
+      options: [{ allowNullableEnum: false }],
+      code: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
+        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
+        if (!theEnum) {
+        }
+      `,
+      errors: [
+        {
+          line: 7,
+          column: 14,
+          messageId: 'conditionErrorNullableEnum',
+          endLine: 7,
+          endColumn: 21,
+        },
+      ],
+      output: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
```

Comment:
Ok great thanks for splitting up the tests! Now that I can read through them a bit more easily, I'm noticing a lack of coverage:
* Enums with implicit values (`enum ExampleEnum { This, That }`)
* Enums with string values (`enum ExampleEnum { This = '', That = "a" }`)
* Mixed enums (`enum ExampleEnum { This = '', That = 0 }`)

Of particular interest is how this deals with string-valued enums with falsy values.

Reply:
@JoshuaKGoldberg 
Thank you!
I added some tests and fixed the code!
https://github.com/typescript-eslint/typescript-eslint/pull/6096/commits/02ec998cf9ab24216852023d4bf9000a1a2e5b92

https://github.com/typescript-eslint/typescript-eslint/pull/6096/commits/39863e08cc9e5df7a3605933622ba453f699a5c3